Google.PowerShell.IntegrationTests/BigQuery/BigQuery.BqTabledata.Tests.ps1

```diff
@@ -0,0 +1,66 @@
+. $PSScriptRoot\..\BigQuery\BqCmdlets.ps1
+$project, $zone, $oldActiveConfig, $configName = Set-GCloudConfig
+
+Describe "New-BqSchema" {
+
+    BeforeAll {
+        $r = Get-Random
+        $datasetName = "pshell_testing_$r"
+        $test_set = New-BqDataset $datasetName
+    }
+
+    It "should create new TableSchema objects"{
+        $schema = New-BqSchema -Name "Title" -Type "STRING"
+        $schema.Fields[0].Name | Should Be "Title"
+        $schema.Fields[0].Type | Should Be "STRING"
+    }
+
+    It "should add fields to existing TableSchema objects"{
+        $schema = New-BqSchema -Name "Title" -Type "STRING"
+        $schema = $schema | New-BqSchema -Name "Author" -Type "STRING"
+        $schema.Fields.Count | Should Be 2
+    }
+
+    It "should handle optional strings correctly"{
+        $schema = New-BqSchema -Name "Title" -Type "STRING" -Description "Test data table" -Mode "REQUIRED"
+        $schema.Fields[0].Description | Should Be "Test data table"
+        $schema.Fields[0].Mode | Should Be "REQUIRED"
+    }
+
+    It "should handle fields / nested structures"{
+        $inner = New-BqSchema -Name "Title" -Type "STRING"
+        $inner = $inner | New-BqSchema -Name "Author" -Type "STRING"
+        $outer = New-BqSchema -Name "Nest" -Type "RECORD" -Fields $inner
+        $outer.Fields[0].Fields.Count | Should Be 2
+    }
+
+    It "should deny invalid types"{
+        { New-BqSchema -Name "Title" -Type "NotAType" -ErrorAction Stop} | Should Throw
+    }
+
+    It "should deny invalid modes"{
+        { New-BqSchema -Name "Title" -Type "STRING" -Mode "NotAMode" -ErrorAction Stop} | Should Throw
```

Comment:
What's the error message here? Same for "invalid types" and "duplicated column names" tests.